api/types/kustomization.go

```diff
@@ -128,23 +134,31 @@ type Kustomization struct {
 // moving content of deprecated fields to newer
 // fields.
 func (k *Kustomization) FixKustomizationPostUnmarshalling() {
-	if k.APIVersion == "" {
-		k.APIVersion = KustomizationVersion
-	}
 	if k.Kind == "" {
 		k.Kind = KustomizationKind
 	}
+	if k.APIVersion == "" {
+		if k.Kind == ComponentKind {
+			k.APIVersion = ComponentVersion
+		} else {
+			k.APIVersion = KustomizationVersion
+		}
+	}
 	k.Resources = append(k.Resources, k.Bases...)
 	k.Bases = nil
 }
 
 func (k *Kustomization) EnforceFields() []string {
 	var errs []string
-	if k.APIVersion != "" && k.APIVersion != KustomizationVersion {
-		errs = append(errs, "apiVersion should be "+KustomizationVersion)
+	if k.Kind != "" && k.Kind != KustomizationKind && k.Kind != ComponentKind {
+		errs = append(errs, "kind should be "+KustomizationKind+" or "+ComponentKind)
+	}
+	requiredVersion := KustomizationVersion
+	if k.Kind == ComponentKind {
+		requiredVersion = ComponentVersion
 	}
-	if k.Kind != "" && k.Kind != KustomizationKind {
-		errs = append(errs, "kind should be "+KustomizationKind)
+	if k.APIVersion != "" && k.APIVersion != requiredVersion {
+		errs = append(errs, "apiVersion should be "+requiredVersion)
```

Comment:
nice check, would you update the message to be:

`"apiVersion for " + k.Kind + " should be " + requiredVersion`

Reply:
Good idea - updated.